Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Utils.cs

```diff
@@ -241,5 +241,56 @@ public static string SearchLatestCompilationDirectory(string debugDirectory)
 
             return depsFile[0].Directory.FullName;            
         }
+
+        /// <summary>
+        /// Returns the Lambda assembly file path that will debugged by the test tool.
+        /// It returns an empty string if no Lambda assembly path is found.
+        /// </summary>
+        /// <param name="debugDirectory">This points to the .../bin/{CONFIGURATION}/{TARGET_FRAMEWORK} directory</param>
+        public static string FindLambdaAssemblyPath(string debugDirectory)
+        {
+            var depsFiles = Directory.GetFiles(debugDirectory, "*.deps.json");
+            if (!depsFiles.Any())
+            {
+                return string.Empty;
+            }
+
+            if (depsFiles.Length == 1)
+            {
+                var depsFilePath = depsFiles.First();
+                var lambdaAssemblyPath = depsFilePath.Substring(0, depsFilePath.Length - ".deps.json".Length) + ".dll";
+                return lambdaAssemblyPath;
+            }
+
+            for (int i = 0; i < depsFiles.Length; i++)
+            {
+                var depsFileName = Path.GetFileName(depsFiles[i]);
+                var projectName = depsFileName.Substring(0, depsFileName.Length - ".deps.json".Length);
+
+                var isRootProject = true;
+
+                for (int j = 0; j < depsFiles.Length; j++)
+                {
+                    if (i == j)
+                        continue;
+                    
+                    var depsFileContent = File.ReadAllText(depsFiles[j]);
+                    if (depsFileContent.Contains($"\"{projectName}\":"))
```

Comment:
You need to parse the JSON file and search each `dependencies` object and see if any of them have `projectName` as a property key. Just doing a string contains will trigger false positives. For example if the Lambda project is called `MyCompany.WebApp.Lambda` but there is another project called `MyCompany.WebApp.Lambda.Common` when seaching the deps file for `MyCompany.WebApp.Lambda` it return back false for being root because you would still find the string `MyCompany.WebApp.Lambda`  inside `MyCompany.WebApp.Lambda.Common`.

Reply:
Make sure the test cases cover this scenario.

Reply:
Eventhough I am not specifically searching for the `dependencies` blob, I am always searching for `"PROJECT_NAME":` (notice the enclosing double-quotes and colon suffix). In the example you provided, we would correctly return `MyCompany.WebApp.Lambda` since we are checking for `"MyCompany.WebApp.Lambda":` in the file contents.

However, to make this more robust, I have updated the PR by parsing all available `dependencies` blob.

The above scenario is already covered by the `MultipleDepsJsonFile` where the dependency graph is as follows:
1. `LambdaDemo` depends on `LambdaDemo.Api1` and `LambdaDemo.Api2`
2. `LambdaDemo.Api1` depends on `LambdaDemo.Api2`